src/server/managers/AppApiManager.ts

```diff
@@ -0,0 +1,156 @@
+
+    constructor(private readonly manager: AppManager) {
+        this.bridge = this.manager.getBridges().getApiBridge();
+        this.accessors = this.manager.getAccessorManager();
+        this.providedApis = new Map<string, Map<string, AppApi>>();
+    }
+
+    /**
+     * Adds an to *be* registered. This will *not register* it with the
+     * bridged system yet as this is only called on an App's
+     * `initialize` method and an App might not get enabled.
+     * When adding an api, it can *not* already exist in the system.
+     *
+     * @param appId the app's id which the api belongs to
+     * @param api the api to add to the system
+     */
+    public addApi(appId: string, api: IApi): void {
+        if (api.endpoints.length === 0) {
+            throw new Error('Invalid Api parameter provided, endpoints must contain, at least, one IApiEndpoint.');
+        }
+
+        const app = this.manager.getOneById(appId);
+        if (!app) {
+            throw new Error('App must exist in order for an api to be added.');
+        }
+
+        // Verify the api's path doesn't exist already
+        if (this.providedApis.get(appId)) {
+            api.endpoints.forEach((endpoint) => {
+                if (this.providedApis.get(appId).has(endpoint.path)) {
+                    throw new PathAlreadyExistsError(endpoint.path);
+                }
+            });
+        }
+
+        if (!this.providedApis.has(appId)) {
+            this.providedApis.set(appId, new Map<string, AppApi>());
+        }
+
+        api.endpoints.forEach((endpoint) => {
+            this.providedApis.get(appId).set(endpoint.path, new AppApi(app, api, endpoint));
+        });
+    }
+
+    /**
+     * Registers all of the api's for the provided app inside
+     * of the bridged system which then enables them.
+     *
+     * @param appId The app's id of which to register it's api's with the bridged system
+     */
+    public registerApis(appId: string): void {
+        if (!this.providedApis.has(appId)) {
+            return;
+        }
+
+        for (const [, apiapp] of this.providedApis.get(appId).entries()) {
+            this.registerApi(appId, apiapp);
+        }
+    }
+
+    /**
+     * Unregisters the api's from the system.
+     *
+     * @param appId the appId for the api's to purge
+     */
+    public unregisterApis(appId: string): void {
+        if (this.providedApis.has(appId)) {
+            this.bridge.unregisterApis(appId);
+
+            this.providedApis.delete(appId);
+        }
+    }
+
+    /**
+     * Executes an App's api.
+     *
+     * @param appId the app which is providing the api
+     * @param path the path to be executed in app's api's
+     * @param request the request data to be evaluated byt the app
+     */
+    public executeApi(appId: string, path: string, request: IApiRequest): Promise<IApiResponse> {
+        const api = this.providedApis.get(appId).get(path);
+
+        if (!api) {
+            return;
```

Comment:
I think we should be returning a response here, even if it's just a 400 bad request. That way the bridged system doesn't have to handle nothing being returned and they actually get back a response when they call this method.